# Copyright 2026 Intrinsic Innovation LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     https://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -15,7 +15,7 @@
 load("@crates_zenoh_c//:defs.bzl", "all_crate_deps")
 load("@rules_cc//cc:defs.bzl", "cc_library")
 load("@rules_rust//cargo:defs.bzl", "cargo_build_script")
-load("@rules_rust//rust:defs.bzl", "rust_binary", "rust_shared_library")
+load("@rules_rust//rust:defs.bzl", "rust_binary", "rust_static_library")
 load("//cbindgen:cbindgen.bzl", "rust_cbindgen_library")
 
 # Transform the opaque types from a .txt to a .rs file.
@@ -40,7 +40,7 @@
 
 # Generate the rust library (name must match the [lib] in Cargo.toml)
 
-rust_shared_library(
+rust_static_library(
     name = "zenohc",
     srcs = glob(
         ["src/**/*.rs"],
@@ -176,6 +176,6 @@
         ],
         "//conditions:default": [],
     }),
-    local_defines = ["ZENOHC_DYN_LIB"],
+
     visibility = ["//visibility:public"],
 )
